node/pkg/driver/node_utils.go

```diff
@@ -515,3 +518,65 @@ func (n NodeUtils) GetTopologyLabels(ctx context.Context, nodeName string) (map[
 	}
 	return topologyLabels, nil
 }
+
+func (n NodeUtils) IsBlock(devicePath string) (bool, error) {
+	var stat unix.Stat_t
+	err := unix.Stat(devicePath, &stat)
+	if err != nil {
+		return false, err
+	}
+	return (stat.Mode & unix.S_IFMT) == unix.S_IFBLK, nil
+}
+
+func (d NodeUtils) GetVolumeStats(path string, isFSVolume bool) (VolumeStatistics, error) {
+	if isFSVolume{
+		return d.getFileSystemVolumeStats(path)
+	} else {
+		return d.getBlockVolumeStats(path)
+	}
+}
+
+func (d NodeUtils) getFileSystemVolumeStats(path string) (VolumeStatistics, error) {
+	statfs := &unix.Statfs_t{}
+	err := unix.Statfs(path, statfs)
+	if err != nil {
+		return VolumeStatistics{}, err
+	}
+
+	availableBytes := int64(statfs.Bavail) * int64(statfs.Bsize)
+	totalBytes := int64(statfs.Blocks) * int64(statfs.Bsize)
+	usedBytes := (int64(statfs.Blocks) - int64(statfs.Bfree)) * int64(statfs.Bsize)
+
+	totalInodes := int64(statfs.Files)
+	availableInodes := int64(statfs.Ffree)
+	usedInodes := totalInodes - availableInodes
+
+	volumeStats := VolumeStatistics{
+		AvailableBytes: availableBytes,
+		TotalBytes:     totalBytes,
+		UsedBytes:      usedBytes,
+
+		AvailableInodes: availableInodes,
+		TotalInodes:     totalInodes,
+		UsedInodes:      usedInodes,
+	}
+
+	return volumeStats, nil
+}
+
+func (d NodeUtils) getBlockVolumeStats(path string) (VolumeStatistics, error) {
+	blockDevice, err := os.Open(path)
```

Comment:
I'm not sure we are suppose to open a user's device. if e.g. there's some limit on the number of open file descriptors, we are limiting the user

Reply:
I found two ways, the first one is to use `blockdev` command as digital ocean used, my fear from this use case is that the command can be changed or the output can be changed.

The second option is what I did here.

What do you think we should do? or there is another way that you know how to do it?

Reply:
done